apps/backend/src/mutations/InstrumentMutations.ts

```diff
@@ -189,10 +189,15 @@ export default class InstrumentMutations {
               );
             })
           );
+          // NOTE: Remove all FAP proposals that are related with the removing instruments
+          await this.fapDataSource.removeProposalsFromFapsByInstrument(

```

Comment:
I don't follow the logic here. Why as part of `assignProposalsToInstrumentsInternal`, a method that sounds like it should only be doing assignments, are we also removing assignments?

Reply:
You are right. I was thinking about this while developing the new feature but will try my best to fix it in some better way.

Reply:
@ACLay I have done some refactoring where I removed that manually cleanup and added better database connections to be able to automatically remove records if proposal is removed from instrument. For example now `fap_proposals` include connection from `instrument_has_proposals` table and if record is removed from `instrument_has_proposals` (proposal is removed from instrument) then proposal is removed from FAP as well automatically in the database level. The same happens with the `technical_review`.